apps/ledger-live-desktop/src/renderer/components/SyncOnboarding/Manual/index.tsx

```diff
@@ -133,18 +178,89 @@ const SyncOnboardingManual = () => {
     }
   }, [device]);
 
+  useEffect(() => {
+    if (deviceOnboardingState?.isOnboarded) {
+      setStepKey(StepKey.SoftwareCheck);
+      return;
+    }
+
+    switch (deviceOnboardingState?.currentOnboardingStep) {
+      case DeviceOnboardingStep.RestoreSeed:
+      case DeviceOnboardingStep.SafetyWarning:
+      case DeviceOnboardingStep.NewDevice:
+      case DeviceOnboardingStep.NewDeviceConfirming:
+        setStepKey(StepKey.Seed);
+        break;
+      case DeviceOnboardingStep.WelcomeScreen:
+      case DeviceOnboardingStep.SetupChoice:
+        setStepKey(StepKey.Paired);
+        break;
+      case DeviceOnboardingStep.Pin:
+        setStepKey(StepKey.Pin);
+        break;
+      default:
+        break;
+    }
+  }, [deviceOnboardingState]);
+
+  useEffect(() => {
+    if (stepKey >= StepKey.SoftwareCheck) {
+      setStopPolling(true);
+    }
+
+    if (stepKey === StepKey.Ready) {
+      setTimeout(() => setStepKey(StepKey.Exit), readyRedirectDelay / 2);
+    }
+
+    if (stepKey === StepKey.Exit) {
+      setTimeout(handleDeviceReady, readyRedirectDelay / 2);
+    }
```

Comment:
Same for `stepKey === StepKey.Exit` on line 170 ?

Reply:
Done